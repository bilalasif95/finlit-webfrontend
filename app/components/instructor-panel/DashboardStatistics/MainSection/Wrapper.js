import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: #255f2b;
  margin: 0 0 4rem;
  .main_section {
    padding: 0 0 4.5rem;
    position: relative;
    h4 {
      font-size: 2rem;
      color: #fff;
      margin: 1rem 0 0;
    }
    .top_links {
      align-items: center;
      display: flex;
      justify-content: flex-start;
      margin: 1.5rem 0 0;
      li {
        display: flex;
        margin: 0 1rem 0 0;
        .link {
          border: 1px solid #fff;
          border-radius: 1.5rem;
          color: #fff;
          font-size: 0.875rem;
          padding: 0.5rem 0.85rem;
          text-decoration: none;
          -webkit-transition: background-color 0.3s ease-out;
          -moz-transition: background-color 0.3s ease-out;
          -o-transition: background-color 0.3s ease-out;
          transition: background-color 0.3s ease-out;
          text-align: center;
          &:hover {
            background-color: #fff;
            color: #255f2b;
          }
        }
        .dropdown-toggle {
          background-color: transparent;
          border: 1px solid #fff;
          border-radius: 1.5rem;
          color: #fff;
          font-size: 0.875rem;
          padding: 0.5rem 0.85rem;
          text-decoration: none;
          -webkit-transition: background-color 0.3s ease-out;
          -moz-transition: background-color 0.3s ease-out;
          -o-transition: background-color 0.3s ease-out;
          transition: background-color 0.3s ease-out;
          &:focus {
            box-shadow: none;
          }
          &:hover {
            background-color: #fff;
            color: #255f2b;
          }
        }
        .dropdown-menu {
          padding: 0;
          li {
            border-bottom: 1px solid #dcdcdc;
            margin: 0;
            &:last-child {
              border-bottom: none;
            }
            .dropdown_link {
              color: #404041 !important;
              font-size: 0.875rem;
              text-decoration: none;
              display: block;
              margin: 0;
              width: 100%;
              padding: 0.5rem 1rem;
              clear: both;
              font-weight: 400;
              text-align: inherit;
              white-space: nowrap;
              background-color: transparent;
              border: 0;
            }
          }
        }
        .dropdoen-item:focus {
          background-color: transparent;
        }
      }
    }
    .main_links {
      align-items: center;
      display: flex;
      justify-content: flex-start;
      margin: 3rem 0 0;
      flex-wrap: wrap;
      width: 100%;
      a {
        border: 1px solid #fff;
        border-radius: 0.25rem;
        color: #fff;
        font-size: 0.875rem;
        margin: 0 1rem 0 0;
        min-width: 200px;
        padding: 0.75rem 1.5rem;
        text-decoration: none;
        text-align: center;
      }
    }
    .stats {
      bottom: -4rem;
      position: absolute;
      width: 100%;
      .single_item {
        align-items: center;
        background-color: #fff;
        border-radius: 0.25rem;
        display: flex;
        justify-content: space-between;
        -webkit-box-shadow: 0px 5px 7px 0px rgba(0, 0, 0, 0.2);
        -moz-box-shadow: 0px 5px 7px 0px rgba(0, 0, 0, 0.2);
        box-shadow: 0px 5px 7px 0px rgba(0, 0, 0, 0.2);
        padding: 2.5rem 1.5rem;
        h2 {
          color: #404041;
          font-weight: 600;
          margin: 0;
          width: 50%;
        }
        p {
          border-left: 1px solid #dcdcdc;
          color: #8b8b8b;
          padding: 0.25rem 0 0.25rem 1rem;
          width: 50%;
        }
      }
    }
  }
  @media (max-width: 1199px) and (min-width: 992px) {
    .main_section {
      .stats {
        .row > * {
          padding-left: 0.5rem !important;
          padding-right: 0.5rem !important;
        }
        .single_item {
          padding: 2.5rem 0.5rem;
          h2 {
            font-size: 1.25rem;
          }
          p {
            padding: 0.25rem 0 0.25rem 0.5rem;
          }
        }
      }
    }
  }
  @media (max-width: 991px) {
    padding: 1.5rem 0 0;
    .main_section {
      padding: 0;
      .main_links {
        a {
          margin: 0 1rem 1rem 0 !important;
        }
      }
      .stats {
        margin: 0 0 -5.5rem;
        position: static !important;
        .single_item {
          margin: 0 0 1.5rem;
          padding: 2.5rem 1rem;
          p {
            padding: 0.25rem 0 0.25rem 1rem;
          }
        }
      }
    }
  }
  @media (max-width: 414px) {
    padding: 0;
    .main_section {
      .top_links {
        flex-wrap: wrap;
        justify-content: space-between;
        li {
          margin: 0 0 1rem;
          width: 47%;
          &:last-child {
            width: 100%;
          }
          .link {
            width: 100%;
          }
          .dropdown {
            width: 100%;
            .show {
              width: 100%;
            }
            .dropdown-toggle {
              align-items: center;
              display: flex;
              justify-content: space-between;
              width: 100%;
            }
          }
        }
      }
      .main_links {
        margin: 1rem 0 0.5rem;
        a {
          margin: 0 0 1rem;
          min-width: 100%;
        }
      }
    }
  }
`;

export default Wrapper;
