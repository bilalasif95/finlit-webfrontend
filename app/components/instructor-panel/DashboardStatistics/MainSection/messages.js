/*
 * Main Section Messages
 *
 * This contains all the text for the Main Section component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.components.MainSection';

export default defineMessages({
  HiEveryone: {
    id: `${scope}.HiEveryone`,
    defaultMessage: 'Hi Li Jordan, good morning',
  },
  MyProfile: {
    id: `${scope}.MyProfile`,
    defaultMessage: `My Profile`,
  },
  MyCourses: {
    id: `${scope}.MyCourses`,
    defaultMessage: `My Courses`,
  },
  AddCourses: {
    id: `${scope}.AddCourses`,
    defaultMessage: `Add Courses`,
  },
  AddWebinar: {
    id: `${scope}.AddWebinar`,
    defaultMessage: `Add Webinar`,
  },
  AddBootCamp: {
    id: `${scope}.AddBootCamp`,
    defaultMessage: `Add BootCamp`,
  },
  AddHackathon: {
    id: `${scope}.AddHackathon`,
    defaultMessage: `Add Hackathon`,
  },
  AddEvents: {
    id: `${scope}.AddEvents`,
    defaultMessage: `Add Events`,
  },
  AddArticles: {
    id: `${scope}.AddArticles`,
    defaultMessage: `Add Articles`,
  },
  Webinars: {
    id: `${scope}.Webinars`,
    defaultMessage: 'Webinars',
  },
  BootCamp: {
    id: `${scope}.BootCamp`,
    defaultMessage: 'BootCamp',
  },
  LiveWebinars: {
    id: `${scope}.LiveWebinars`,
    defaultMessage: 'Live Webinars',
  },
  Hackathon: {
    id: `${scope}.Hackathon`,
    defaultMessage: 'Hackathon',
  },
  FreeGiftsRewards: {
    id: `${scope}.FreeGiftsRewards`,
    defaultMessage: 'Free Gifts & Rewards',
  },
  Bootcamps: {
    id: `${scope}.Bootcamps`,
    defaultMessage: `Bootcamps`,
  },
  Courses: {
    id: `${scope}.Courses`,
    defaultMessage: `Courses`,
  },
  Hackathons: {
    id: `${scope}.Hackathons`,
    defaultMessage: `Hackathons`,
  },
});
