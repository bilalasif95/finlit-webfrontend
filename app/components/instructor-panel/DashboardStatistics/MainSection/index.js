import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';
import {
  Container,
  Row,
  Col,
  Dropdown,
  DropdownToggle,
  DropdownMenu,
} from 'reactstrap';
import { Link } from 'react-router-dom';
import Wrapper from './Wrapper';
import messages from './messages';

function MainSection(props) {
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const [greetingMessage, setGreetingMessage] = useState('');
  const [userObj, setUserObj] = useState({});

  const toggle = () => setDropdownOpen(prevState => !prevState);
  let statistics = props.statistics;

  React.useEffect(() => {
    const userObj = JSON.parse(localStorage.getItem('userInfo') || '');
    console.log(userObj.firstName);
    setUserObj(userObj);
    const time = new Date().getHours();
    let greeting;
    if (time < 10) {
      greeting = 'Good Morning';
    } else if (time < 20) {
      greeting = 'Good Afternoon';
    } else {
      greeting = 'Good Evening';
    }
    setGreetingMessage(greeting);
  }, []);

  return (
    <Wrapper>
      <div className="custom_container">
        <Container fluid>
          <div className="main_section">
            <h4>
              Hi {userObj.firstName} {userObj.lastName}, {greetingMessage}
            </h4>
            <ul className="top_links">
              <li>
                <Link className="link" to="/my_profile">
                  <FormattedMessage {...messages.MyProfile} />
                </Link>
              </li>
              {userObj &&
              userObj.roles &&
              userObj.roles[0] &&
              userObj.roles[0].roleName === 'Instructor' ? (
                <>
                  <li>
                    <Link className="link" to="/courses_list">
                      <FormattedMessage {...messages.MyCourses} />
                    </Link>
                  </li>
                  <li>
                    <Dropdown isOpen={dropdownOpen} toggle={toggle}>
                      <DropdownToggle caret>Add Category</DropdownToggle>
                      <DropdownMenu>
                        <ul>
                          <li>
                            <Link className="dropdown_link" to="/add_course">
                              <FormattedMessage {...messages.AddCourses} />
                            </Link>
                          </li>
                          <li>
                            <Link className="dropdown_link" to="/add_webinar">
                              <FormattedMessage {...messages.AddWebinar} />
                            </Link>
                          </li>
                          <li>
                            <Link className="dropdown_link" to="/add_bootcamp">
                              <FormattedMessage {...messages.AddBootCamp} />
                            </Link>
                          </li>
                          <li>
                            <Link className="dropdown_link" to="/add_hackathon">
                              <FormattedMessage {...messages.AddHackathon} />
                            </Link>
                          </li>
                          <li>
                            <Link className="dropdown_link" to="/add_article">
                              <FormattedMessage {...messages.AddArticles} />
                            </Link>
                          </li>
                        </ul>
                      </DropdownMenu>
                    </Dropdown>
                  </li>
                </>
              ) : (
                <></>
              )}
            </ul>
            <div className="main_links">
              {userObj &&
              userObj.roles &&
              userObj.roles[0] &&
              userObj.roles[0].roleName === 'Instructor' ? (
                <>
                  <Link to="/webinars_list">
                    <FormattedMessage {...messages.Webinars} />
                  </Link>
                  <Link to="/bootcamps_list">
                    <FormattedMessage {...messages.BootCamp} />
                  </Link>
                  <Link to="/hackathons_list">
                    <FormattedMessage {...messages.Hackathon} />
                  </Link>
                </>
              ) : (
                <>
                  <Link to="/webinar_list">
                    <FormattedMessage {...messages.Webinars} />
                  </Link>
                  <Link to="/bootcamp_list">
                    <FormattedMessage {...messages.BootCamp} />
                  </Link>
                  <Link to="/hackathon_list">
                    <FormattedMessage {...messages.Hackathon} />
                  </Link>
                </>
              )}
            </div>
            {userObj &&
            userObj.roles &&
            userObj.roles[0] &&
            userObj.roles[0].roleName === 'Instructor' ? (
              <>
                <div className="stats">
                  <Row>
                    <Col lg={3} md={6} sm={6} xs={12}>
                      <div className="single_item">
                        <h2>{statistics.totalBootcamps}</h2>
                        <span className="vl" />
                        <p>
                          <FormattedMessage {...messages.Bootcamps} />
                        </p>
                      </div>
                    </Col>
                    <Col lg={3} md={6} sm={6} xs={12}>
                      <div className="single_item">
                        <h2>{statistics.totalWebinars}</h2>
                        <span />
                        <p>
                          <FormattedMessage {...messages.Webinars} />
                        </p>
                      </div>
                    </Col>
                    <Col lg={3} md={6} sm={6} xs={12}>
                      <div className="single_item">
                        <h2>{statistics.totalCourses}</h2>
                        <span />
                        <p>
                          <FormattedMessage {...messages.Courses} />
                        </p>
                      </div>
                    </Col>
                    <Col lg={3} md={6} sm={6} xs={12}>
                      <div className="single_item">
                        <h2>{statistics.totalHackathons}</h2>
                        <span />
                        <p>
                          <FormattedMessage {...messages.Hackathons} />
                        </p>
                      </div>
                    </Col>
                  </Row>
                </div>
              </>
              ) : (
              <></>
            )}
          </div>
        </Container>
      </div>
    </Wrapper>
  );
}

export default MainSection;
