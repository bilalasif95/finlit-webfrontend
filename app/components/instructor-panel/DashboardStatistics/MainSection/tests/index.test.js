import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';

import MainSection from '../index';
import configureStore from '../../../../../configureStore';

describe('<MainSection />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <BrowserRouter>
              <MainSection />
            </BrowserRouter>
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
