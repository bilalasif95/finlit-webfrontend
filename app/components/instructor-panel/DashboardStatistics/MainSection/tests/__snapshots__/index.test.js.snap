// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSection /> should render and match the snapshot 1`] = `
<div
  className="Wrapper-sc-1hjyhky-0 kVAarz"
>
  <div
    className="custom_container"
  >
    <div
      className="container-fluid"
    >
      <div
        className="main_section"
      >
        <h4>
          <span>
            Hi Li Jordan, good morning
          </span>
        </h4>
        <ul
          className="top_links"
        >
          <li>
            <a
              className="link"
              href="/my_profile"
              onClick={[Function]}
            >
              <span>
                My Profile
              </span>
            </a>
          </li>
          <li>
            <a
              className="link"
              href="/courses_list"
              onClick={[Function]}
            >
              <span>
                My Courses
              </span>
            </a>
          </li>
          <li>
            <div
              className="dropdown"
              onKeyDown={[Function]}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label={null}
                className="dropdown-toggle btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Add Category
              </button>
              <div
                aria-hidden={true}
                className="dropdown-menu"
                role="menu"
                tabIndex="-1"
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/add_course"
                    onClick={[Function]}
                  >
                    <span>
                      Add Courses
                    </span>
                  </a>
                </button>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/add_webinar"
                    onClick={[Function]}
                  >
                    <span>
                      Add Webinar
                    </span>
                  </a>
                </button>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/add_bootcamp"
                    onClick={[Function]}
                  >
                    <span>
                      Add BootCamp
                    </span>
                  </a>
                </button>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/add_hackathon"
                    onClick={[Function]}
                  >
                    <span>
                      Add Hackathon
                    </span>
                  </a>
                </button>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/"
                    onClick={[Function]}
                  >
                    <span>
                      Add Events
                    </span>
                  </a>
                </button>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  <a
                    className="dropdown_link"
                    href="/"
                    onClick={[Function]}
                  >
                    <span>
                      Add Articles
                    </span>
                  </a>
                </button>
              </div>
            </div>
          </li>
        </ul>
        <div
          className="main_links"
        >
          <a
            href="/webinars_list"
            onClick={[Function]}
          >
            <span>
              Webinars
            </span>
          </a>
          <a
            href="/bootcamp_list"
            onClick={[Function]}
          >
            <span>
              BootCamp
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <span>
              Live Webinars
            </span>
          </a>
          <a
            href="/hackathon_list"
            onClick={[Function]}
          >
            <span>
              Hackathon
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <span>
              Free Gifts & Rewards
            </span>
          </a>
        </div>
        <div
          className="stats"
        >
          <div
            className="row"
          >
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-3"
            >
              <div
                className="single_item"
              >
                <h2>
                  510$
                </h2>
                <span
                  className="vl"
                />
                <p>
                  <span>
                    Total Revenue
                  </span>
                </p>
              </div>
            </div>
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-3"
            >
              <div
                className="single_item"
              >
                <h2>
                  40+
                </h2>
                <span />
                <p>
                  <span>
                    Webinars
                  </span>
                </p>
              </div>
            </div>
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-3"
            >
              <div
                className="single_item"
              >
                <h2>
                  25+
                </h2>
                <span />
                <p>
                  <span>
                    Courses
                  </span>
                </p>
              </div>
            </div>
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-3"
            >
              <div
                className="single_item"
              >
                <h2>
                  510$
                </h2>
                <span />
                <p>
                  <span>
                    Events
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
